docs/tools/plist.md

```diff
@@ -0,0 +1,301 @@
+    <string>event</string>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>1</integer>
+      <key>file</key>
+      <integer>0</integer>
+      <key>line</key>
+      <integer>19</integer>
+    </dict>
+    <key>message</key>
+    <string>First bug path event</string>
+  </dict>
+  <dict>
+    <key>depth</key>
+    <integer>0</integer>
+    <key>kind</key>
+    <string>event</string>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>1</integer>
+      <key>file</key>
+      <integer>1</integer>
+      <key>line</key>
+      <integer>3</integer>
+    </dict>
+    <key>message</key>
+    <string>Second bug path event</string>
+  </dict>
+</array>
+```
+
+UI Representation:
+
+![Web interface showing bug path events](images/bug_path_events.png)
+
+Navigating through Bug Path Events is possible by using “Previous” and “Next” 
+buttons. This makes it easy to step through the logical flow that triggered 
+the bug - even when it spans across multiple files.
+
+If two consecutive bug path events occur in the same file, the UI will draw an 
+arrow in the code viewer pane, pointing from one event to the next.
+These arrows help visually trace how the issue progresses through the code. 
+
+Arrow visibility can be turned on/off in the UI settings.
+
+### 2. `notes` — Notes
+
+```xml
+<key>notes</key>
+<array>
+  <dict>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>15</integer>
+      <key>file</key>
+      <integer>0</integer>
+      <key>line</key>
+      <integer>3</integer>
+    </dict>
+    <key>message</key>
+    <string>This is a note</string>
+  </dict>
+</array>
+```
+
+UI Representation:
+
+![Web interface showing a note](images/note.png)
+
+### 3. `macro_expansions` — Macro Expansions
+
+```xml
+<key>macro_expansions</key>
+<array>
+  <dict>
+    <key>expansion</key>
+    <string>Macro expansion message</string>
+    <key>location</key>
+    <dict>
+      <key>col</key>
+      <integer>15</integer>
+      <key>file</key>
+      <integer>1</integer>
+      <key>line</key>
+      <integer>3</integer>
+    </dict>
+    <key>name</key>
+    <string>Macro name which will be expanded.</string>
+  </dict>
+</array>
+```
+
+UI Representation:
+
+![Web interface showing a macro expansion](images/macro_expansion.png)
+
+### 4. `report-annotation` — Report Annotations
+
+Report Annotations are a special kind of metadata associated with reports in 
+CodeChecker. They were originally introduced to incorporate dynamic analysis 
+tools better, where additional contextual data is useful. However, 
+Report Annotations can be associated with any Report, including the ones
+generated by static analysis tools.
+
+A report-annotation may include:
+- **`timestamp`** (`date-time`, ISO 8601): Timestamp when the issue was generated, recorded, or annotated
+- **`testcase`** (`string`): Customizable name of the Testcase
+- **`chronological_order`** (`int`): Sequence number of the issue
```

Comment:
In dynamic analysis the order of issues are important. With the chronological order field, the order of dynamic anlaysis events can be recorded.